psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java

```diff
@@ -0,0 +1,513 @@
+
+        validateCaseIdentification(model.getCaseIdentification());
+        validateBuses(model.getBuses(), buses);
+        validateLoads(model.getLoads(), buses);
+        validateFixedShunts(model.getFixedShunts(), buses);
+        validateGenerators(model.getBuses(), model.getGenerators(), buses);
+        validateNonTransformerBranches(model.getNonTransformerBranches(), buses);
+        validateTransformers(model.getTransformers(), buses);
+
+        validateSwitchedShunts(model.getSwitchedShunts(), buses, psseVersion);
+    }
+
+    private static Map<Integer, List<Integer>> generateBuses(List<PsseBus> psseBuses) {
+        Map<Integer, List<Integer>> buses = new HashMap<>();
+        for (int i = 0; i < psseBuses.size(); i++) {
+            buses.computeIfAbsent(psseBuses.get(i).getI(), k -> new ArrayList<>()).add(i);
+        }
+        return buses;
+    }
+
+    private void validateCaseIdentification(PsseCaseIdentification caseIdentification) {
+        if (caseIdentification.getSbase() <= 0.0) {
+            warnings.add(String.format(Locale.US, "CaseIdentification: Unexpected Sbase: %.2f", caseIdentification.getSbase()));
+            validCase = false;
+        }
+        if (caseIdentification.getBasfrq() <= 0.0) {
+            warnings.add(String.format(Locale.US, "CaseIdentification: Unexpected Basfrq: %.2f", caseIdentification.getBasfrq()));
+            validCase = false;
+        }
+    }
+
+    private void validateBuses(List<PsseBus> psseBuses, Map<Integer, List<Integer>> buses) {
+        for (Map.Entry<Integer, List<Integer>> entry : buses.entrySet()) {
+            if (entry.getValue().size() != 1) {
+                warnings.add(String.format("Bus: %d defined multiple times (%d)", entry.getKey(), entry.getValue().size()));
+                validCase = false;
+            }
+        }
+
+        for (int i = 0; i < psseBuses.size(); i++) {
+            PsseBus psseBus = psseBuses.get(i);
+            if (psseBus.getI() < 1 || psseBus.getI() > 999997) {
+                warnings.add(String.format("Bus: Unexpected I: %d", psseBus.getI()));
+                validCase = false;
+            }
+            if (psseBus.getBaskv() < 0.0) {
+                warnings.add(String.format(Locale.US, "Bus: %d Unexpected Baskv: %.2f", psseBus.getI(), psseBus.getBaskv()));
+                validCase = false;
+            }
+        }
+    }
+
+    private void validateLoads(List<PsseLoad> loads, Map<Integer, List<Integer>> buses) {
+        Map<String, List<String>> busesLoads = new HashMap<>();
+
+        for (int i = 0; i < loads.size(); i++) {
+            PsseLoad load = loads.get(i);
+            if (!buses.containsKey(load.getI())) {
+                warnings.add(String.format("Load: Unexpected I: %d", load.getI()));
+                validCase = false;
+            }
+            addBusesMap(busesLoads, load.getI(), load.getId());
+        }
+
+        Map<String, List<String>> duplicatedBusesLoads = getDuplicates(busesLoads);
```

Comment:
this check for duplicates is repeated in loads, generators, shunts, ... it can be moved to a method that accepts the name of the type as a parameter ... like `checkDuplicates("loads", busesLoads)`

Reply:
Done